pkg/eval/mods/file/file.go

```diff
@@ -124,3 +127,55 @@ func prclose(p vals.Pipe) error {
 func pwclose(p vals.Pipe) error {
 	return p.WriteEnd.Close()
 }
+
+//elvdoc:fn truncate
+//
+// ```elvish
+// file:truncate $filename $size
+// ```
+// Truncate changes the size of
+// the named file. If the file is
+// a symbolic link, it changes the
+// size of the link's target.
+//
+// If there is an error it will
+// be of type *PathError.
+
+func truncate(name string, size vals.Num) error {
+	var offset int64
+	switch size.(type) {
+	case int:
+		offset = int64(size.(int))
+		if offset < 0 {
+			return errs.BadValue{
+				What:   "Numeric input for the function is wrong",
+				Valid:  "positive 64 bit integer",
+				Actual: "Number less than 0",
+			}
```

Comment:
This should use errs.OutOfRange. The multiple checks against negative values should also be merged together.